src/Servers/Kestrel/shared/test/Http3/Http3InMemory.cs

```diff
@@ -205,20 +201,26 @@ public void AdvanceTime(TimeSpan timeSpan)
 
         var timeProvider = _mockTimeProvider;
         var endTime = timeProvider.GetTimestamp() + timeSpan.ToTicks(timeProvider.TimestampFrequency);
+        var interval = Heartbeat.Interval.ToTicks(timeProvider.TimestampFrequency);
 
-        while (timeProvider.GetTimestamp() + Heartbeat.Interval.ToTicks(timeProvider.TimestampFrequency) < endTime)
+        while (timeProvider.GetTimestamp() + interval < endTime)
         {
-            timeProvider.Advance(Heartbeat.Interval);
-            _timeoutControl.Tick(timeProvider.GetTimestamp());
+            TriggerTick(timeProvider.GetTimestamp() + interval);
         }
 
-        timeProvider.SetTimestamp(endTime);
-        _timeoutControl.Tick(timeProvider.GetTimestamp());
+        TriggerTick(endTime);
     }
 
     public void TriggerTick(DateTimeOffset now)
     {
         _mockTimeProvider.SetUtcNow(now);
+        Connection?.Tick(_mockTimeProvider.GetTimestamp());
+    }
+
+    public void TriggerTick(long now)
```

Comment:
What is the long? I think variables like this need to say what the value is, e.g. nowTimestamp, or just timestamp